File: packages/main-api/src/features/spaces/artworks/models/space_artwork.rs
use crate::{types::*, utils::time::get_now_timestamp_millis};
use bdk::prelude::*;

#[derive(
    Debug,
    Clone,
    serde::Serialize,
    serde::Deserialize,
    DynamoEntity,
    Default,
    JsonSchema,
    aide::OperationIo,
)]
#[dynamo(table = "main")]
pub struct SpaceArtwork {
    pub pk: Partition,
    pub sk: EntityType,

    pub created_at: i64,
    pub updated_at: i64,

    pub contract_address: String,
    pub metadata_uri: String,
    pub metadata: String,
    pub owner_evm_address: String,
}

impl SpaceArtwork {
    pub fn new(
        space_pk: Partition,
        contract_address: String,
        metadata_uri: String,
        metadata: String,
        owner_evm_address: String,
    ) -> Self {
        let now = get_now_timestamp_millis();
        Self {
            pk: space_pk,
            sk: EntityType::SpaceArtwork,
            created_at: now,
            updated_at: now,
            contract_address,
            metadata_uri,
            metadata,
            owner_evm_address,
        }
    }
}

File: ts-packages/web/src/features/spaces/art-nfts/types/space-artwork.ts
import SpaceArtworkResponse from '../dto/space-artwork-response';

export default class SpaceArtwork {
  public pk: string;
  public sk: string;
  public createdAt: number;
  public updatedAt: number;
  public contractAddress: string;
  public metadataUri: string;
  public metadata: string;

  constructor(data: SpaceArtworkResponse) {
    this.pk = data.pk;
    this.sk = data.sk;
    this.createdAt = data.created_at;
    this.updatedAt = data.updated_at;
    this.contractAddress = data.contract_address;
    this.metadataUri = data.metadata_uri;
    this.metadata = data.metadata;
  }
}
